Cubical/Homotopy/Connected.agda

```diff
@@ -93,3 +186,127 @@ isHLevelConnectedPoint n connA a₀ a =
+    Iso.rightInv theIso f = funExt λ y → sym (helper f y)
+    Iso.leftInv theIso _ = refl
+
+connectedTruncIso : ∀ {ℓ} {A B : Type ℓ} (n : ℕ) (f : A → B)
+                   → isHLevelConnectedFun n f
+                   → Iso (hLevelTrunc n A) (hLevelTrunc n B)
+connectedTruncIso {A = A} {B = B} zero f con = g
+  where
+  g : Iso (hLevelTrunc zero A) (hLevelTrunc zero B)
+  Iso.fun g = λ _ → fst (isOfHLevelTrunc 0)
+  Iso.inv g = λ _ → fst (isOfHLevelTrunc 0)
+  Iso.rightInv g = λ b → isOfHLevelTrunc 0 .snd b
+  Iso.leftInv g = λ b → isOfHLevelTrunc 0 .snd b
+connectedTruncIso {A = A} {B = B} (suc n) f con = g
+  where
+  back : B → hLevelTrunc (suc n) A
+  back y = map fst ((con y) .fst)
+
+  backSection :  (b : B) → Path (hLevelTrunc (suc n) B)
+                                 (Trunc.rec (isOfHLevelTrunc (suc n))
+                                            (λ a → ∣ f a ∣)
+                                            (Trunc.rec {n = suc n }
+                                                       {B = hLevelTrunc (suc n) A} (isOfHLevelTrunc (suc n)) back ∣ b ∣))
+                               ∣ b ∣
+  backSection b = helper (λ p → map f p ≡ ∣ b ∣)
+                         (λ x → (isOfHLevelSuc (suc n) (isOfHLevelTrunc (suc n)))
+                                   (map f (map fst x)) ∣ b ∣)
+                         (λ a p → cong ∣_∣ p)
+                         (fst (con b))
+    where
+    helper : ∀ {ℓ ℓ' ℓ''} {A : Type ℓ} {B : A → Type ℓ'} (P : hLevelTrunc (suc n) A → Type ℓ'')
+           → ((x : hLevelTrunc (suc n) (Σ A B)) → isOfHLevel (suc n) (P (map fst x)))
+           → ((a : A) (b : B a) → P ∣ a ∣)
+           → (p : hLevelTrunc (suc n) (Σ A B))
+           →  P (map fst p)
+    helper P hlev pf = Trunc.elim hlev λ pair → pf (fst pair) (snd pair)
+
+  backRetract : (a : A) → map fst (con (f a) .fst) ≡ ∣ a ∣
+  backRetract a = cong (map fst) (con (f a) .snd ∣ a , refl ∣)
+
+  g : Iso (hLevelTrunc (suc n) A) (hLevelTrunc (suc n) B)
+  Iso.fun g = map f
+  Iso.inv g = Trunc.rec (isOfHLevelTrunc _) back
+  Iso.leftInv g = Trunc.elim (λ x → isOfHLevelPath (suc n) (isOfHLevelTrunc _) _ _)
+                               λ a → cong (map fst) (con (f a) .snd  ∣ a , refl ∣)
+  Iso.rightInv g = Trunc.elim (λ x → isOfHLevelPath (suc n) (isOfHLevelTrunc _) _ _)
+                              backSection
+
+connectedTruncIso2 : ∀ {ℓ} {A B : Type ℓ} (n m : ℕ) (f : A → B)
+                   → Σ[ x ∈ ℕ ] x + n ≡ m
+                   → isHLevelConnectedFun m f
+                   → Iso (hLevelTrunc n A) (hLevelTrunc n B)
+connectedTruncIso2 {A = A} {B = B} n m f (x , pf) con =
+  connectedTruncIso n f (isConnectedSubtr n x f (transport (λ i → isHLevelConnectedFun (pf (~ i)) f) con))
+
+connectedTruncEquiv : ∀ {ℓ} {A B : Type ℓ} (n : ℕ) (f : A → B)
+                   → isHLevelConnectedFun n f
+                   → hLevelTrunc n A ≃ hLevelTrunc n B
+connectedTruncEquiv {A = A} {B = B} n f con = isoToEquiv (connectedTruncIso n f con)
+
```

Comment:
@mortberg @guillaumebrunerie This is the only comment I havent seen to because I am not really sure where to put them. Is it ok to leave this as it is for now, and maybe split things up when we get more theorems in there?

And sorry for being so slow with fixing this PR. I got so caught up in other things..

Reply:
Yes, I think it’s fine to leave it as is.

Reply:
Yeah, you can add a TODO about reorganizing things so that we don't forget

Reply:
Alright, added. I think that's all the requested changes.